newrelic/python_agent.go

```diff
@@ -0,0 +1,71 @@
+package newrelic
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/buildpacks/libcnb"
+	"github.com/paketo-buildpacks/libpak"
+	"github.com/paketo-buildpacks/libpak/bard"
+	"github.com/paketo-buildpacks/libpak/effect"
+	"github.com/paketo-buildpacks/libpak/sherpa"
+)
+
+type PythonAgent struct {
+	BuildpackPath    string
+	ApplicationPath  string
+	Executor         effect.Executor
+	LayerContributor libpak.DependencyLayerContributor
+	Logger           bard.Logger
+}
+
+func NewPythonAgent(applicationPath string, buildpackPath string, dependency libpak.BuildpackDependency, cache libpak.DependencyCache) (PythonAgent, libcnb.BOMEntry) {
+	contributor, entry := libpak.NewDependencyLayer(dependency, cache, libcnb.LayerTypes{
+		Launch: true,
+	})
+	return PythonAgent{
+		ApplicationPath:  applicationPath,
+		BuildpackPath:    buildpackPath,
+		Executor:         effect.NewExecutor(),
```

Comment:
It's easy enough to create an executor here, but then it's impossible to mock. I would strongly suggest you take the executor as an argument to NewPythonAgent, that way when you're testing you can pass in a mock executor and mock the calls. We can't have the test actually run `pip`.

Reply:
Done